code/iff_defs/iff_defs.cpp

```diff
@@ -289,30 +399,56 @@ void iff_init()
 		// begin reading data
 		while (required_string_either("#End", "$IFF Name:"))
 		{
-			iff_info *iff;
-			int cur_iff;
-
-			// add new IFF
-			cur_iff = (int) Iff_info.size();
-			Iff_info.push_back(iff_info());
-			attack_names.emplace_back();
-			observed_color_table.emplace_back();
-			iff = &Iff_info[cur_iff];
-
-
-			// get required IFF info ----------------------------------------------
 
 			// get the iff name
 			required_string("$IFF Name:");
-			stuff_string(iff->iff_name, F_NAME, NAME_LENGTH);
+			stuff_string(ifft.iff_name, F_NAME, NAME_LENGTH);
+			mprintf(("IFFs got to name %s\n", ifft.iff_name));
+			if (optional_string("+nocreate")) {
+				if (!Parsing_modular_table) {
+					Warning(LOCATION, "+nocreate flag used for iff in non-modular table\n");
+				}
+				create_if_not_found = false;
+			}
+
+			// Does this iff exist already?
+			// If so, load this new info into it
+			iffp = get_iff_pointer(ifft.iff_name);
+			if (iffp != nullptr) {
+				if (!Parsing_modular_table) {
+					error_display(1,
+						"Error:  IFF %s already exists.  All IFF names must be unique.",
+						ifft.iff_name);
+				}
+			} else {
+				// Don't create iff if it has +nocreate and is in a modular table.
+				if (!create_if_not_found && Parsing_modular_table) {
+					if (!skip_to_start_of_string_either("$IFF Name:", "#End")) {
+						error_display(1, "Missing [#End] or [$IFF Name] after IFF %s", ifft.iff_name);
+					}
+					continue;
+				}
+				Iff_info.push_back(ifft);
+				attack_names.emplace_back();
+				observed_color_table.emplace_back();
+				iffp = &Iff_info[Iff_info.size() - 1];
+				//Initialize this to white for new IFFs
+				iffp->color_index = iff_init_color(255, 255, 255);
+				//Make sure flags are reset for new IFFs
+				iffp->default_parse_flags.reset();
+			}
+
+			cur_iff = get_iff_position(ifft.iff_name);
 
 			// get the iff color
-			if (check_for_string("$Colour:"))
-				required_string("$Colour:");
-			else
-				required_string("$Color:");
-			stuff_int_list(rgb, 3, RAW_INTEGER_TYPE);
-			iff->color_index = iff_init_color(rgb[0], rgb[1], rgb[2]);
+			if (optional_string("$Colour:")) {
+				stuff_int_list(rgb, 3, RAW_INTEGER_TYPE);
+				iffp->color_index = iff_init_color(rgb[0], rgb[1], rgb[2]);
+			}
+			if (optional_string("$Color:")) {
+				stuff_int_list(rgb, 3, RAW_INTEGER_TYPE);
+				iffp->color_index = iff_init_color(rgb[0], rgb[1], rgb[2]);
+			}
```

Comment:
This should probably be an optional string either to avoid code repetition here. Also, what will this do in a modular table where color isn't defined? I think this also needs an else if not nocreate, then error:
```suggestion
			if (optional_string_either("$Colour:", "$Color:") != -1) { 
				stuff_int_list(rgb, 3, RAW_INTEGER_TYPE);
				iffp->color_index = iff_init_color(rgb[0], rgb[1], rgb[2]);
			}
			else if (create_if_not_found) {
				error_display(1, "Error:  IFF %s must be specified a color.", ifft.iff_name);
			}
```

Reply:
I think I initialized the color to white to make sure it has data if this option was skipped in a modular table. Will double check when I make the other changes.

Reply:
It sets color as white when a new IFF is being parsed at line 436 and then overwrites that color here if it's defined. All other changes made.

Reply:
I see, should be fine if documented as new default then.